File: dotnet/FinkiRasporedi/Models/Domain/Course.cs
using FinkiRasporedi.Models.Domain;
using System.ComponentModel.DataAnnotations;

namespace FinkiRasporedi.Models.Domain
{
    public class Course
    {
        [Key]
        public string Id { get; set; }

        public virtual Semester Semester { get; set; }

        public virtual Subject Subject { get; set; }

    }
}

File: flutter/lib/domain/models/course.dart
import 'package:flutter_app/domain/models/semester.dart';
import 'package:flutter_app/domain/models/subject.dart';

class Course {
  String id;
  Semester semester;
  Subject subject;

  Course({
    required this.id,
    required this.semester,
    required this.subject,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['id'],
      semester: Semester.fromJson(json['semester']),
      subject: Subject.fromJson(json['subject']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'semester': semester,
      'subject': subject.toJson(),
    };
  }
}
